### PR TITLE
Update default version number to 15.1.0

### PR DESCRIPTION
**Pull request by smadala**
- As 15.0.0 RTM has released, the next version from master will be 15.1.0.
- Make Nuget packages and VSIX will have same version.

### CODE REVIEW

**Review by codito — APPROVED** [review]
Please validate if bindings redirects are in place for `15.1.0` in net46 target.

**Review by Faizan2304 — APPROVED** [review]
Please validate this new Test.Net.Sdk with old test platform

File: scripts/build.ps1 [hunk 1]
```diff
@@ -569,3 +553,28 @@ else
...
+    # Execute build
+    $timer = Start-Timer
+    Write-Log "Build started: args = '$args'"
+    Write-Log "Test platform environment variables: "
+    Get-ChildItem env: | Where-Object -FilterScript { $_.Name.StartsWith("TP_") } | Format-Table
+    Write-Log "Test platform build variables: "
+    Get-Variable | Where-Object -FilterScript { $_.Name.StartsWith("TPB_") } | Format-Table
+    Install-DotNetCli
+    Restore-Package
+    #Update-LocalizedResources
```
Comment (codito) [inline scripts/build.ps1]: Should not be commented.

File: scripts/build/TestPlatform.Dependencies.props [hunk 2]
```diff
@@ -2,7 +2,7 @@
 <Project ToolsVersion="4.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
   <PropertyGroup>
     <!-- Name of the elements must be in sync with test\Microsoft.TestPlatform.TestUtilities\IntegrationTestBase.cs -->
-    <NETTestSdkPreviousVersion>15.0.0-preview-20170106-08</NETTestSdkPreviousVersion>
+    <NETTestSdkPreviousVersion>15.0.0</NETTestSdkPreviousVersion>
 
     <MSTestFrameworkVersion>1.0.8-rc2</MSTestFrameworkVersion>
```
Comment (codito) [inline scripts/build/TestPlatform.Dependencies.props]: Update mstest please.

File: test/Microsoft.TestPlatform.SmokeTests/DataCollectorTests.cs [hunk 3]
```diff
@@ -39,13 +44,19 @@ private static void ValidateInProcDataCollectionOutput()
 
         private string GetInProcDataCollectionRunsettingsFile()
         {
-            var runSettings = Path.Combine(Path.GetDirectoryName(this.GetSampleTestAssembly()), "runsettingstest.runsettings");
-            var testEnvironment = new IntegrationTestEnvironment();
+            var runSettings = Path.Combine(Path.GetDirectoryName(testEnvironment.GetTestAsset(DataCollectorTests.InProDataCollectorTestProject)), "runsettingstest.runsettings");
             var inprocasm = testEnvironment.GetTestAsset("SimpleDataCollector.dll");
+#if !NET46
```
Comment (codito) [inline test/Microsoft.TestPlatform.SmokeTests/DataCollectorTests.cs]: Please use AssemblyName.GetAssemblyName.

**Review by codito — APPROVED** [review]

**Pull request merged**